doc/cookbook/namedRoutes/NamedRoutes.lhs

```diff
@@ -0,0 +1,371 @@
+# NamedRoutes - Using records to define APIs
+
+*Available in Servant 0.19 or higher*
+
+Servant offers a very natural way of constructing APIs with nested records, called `NamedRoutes`.
+
+This cookbook explains how to implement such nested-record-APIs using `NamedRoutes` through the example of a Movie Catalog.
+
+First, we start by constructing the domain types of our Movie Catalog.
+After, we show you how to implement the API type with the NamedRoutes records.
+Lastly, we make a Server and a Client out of the API type.
+
+However, it should be understood that this cookbook does _not_ dwell on the built-in servant combinators as the [<Structuring APIs> cookbook ](<https://docs.servant.dev/en/stable/cookbook/structuring-apis/StructuringApis.html>) already covers that angle.
+
+## Why would I want to use `NamedRoutes` over the alternative `:<|>` operator?
+
+With `NamedRoutes`, we don’t need to care about the declaration order of the endpoints.
```

Comment:
I'd say just go straight into "this gives better errors". And perhaps you can indeed mention that we can afford not to care about order in the "record API" scenario, by using record syntax to construct the record of handlers, for example. `ConstructorName f1 f2` (order matters) vs `ConstructorName { field2 = f2, field1 = f1 }` (order doesn't matter, names do). But only after the better, "per-field" errors, IMO.